### PR TITLE
Disable `AX_PTHREAD` for MingW/MSYS builds.

### PR DESCRIPTION
**Pull request by utoni**
Please sign (check) the below before submitting the Pull Request:

- [x] I have signed the ntop Contributor License Agreement at https://github.com/ntop/legal/blob/main/individual-contributor-licence-agreement.md
- [x] I have read the contributing guide lines at https://github.com/ntop/nDPI/blob/dev/CONTRIBUTING.md
- [x] I have updated the documentation (in doc/) to reflect the changes made (if applicable)

Link to the related [issue](https://github.com/ntop/nDPI/issues):

Describe changes:

### CODE REVIEW

**Comment by IvanNardi** [comment]
@utoni, is this PR about the failing CI jobs on windows?

**Comment by utoni** [comment]
It is. Somehow `AX_PTHREAD` fails to find `winpthreads`.

**Review by IvanNardi — APPROVED** [review]
Thanks for fixing this issue!
BTW, do you have any idea why it stops working right now? What changes?

File: configure.ac [hunk 1]
```diff
@@ -178,11 +178,14 @@ if ! test "${with_only_libndpi+set}" = set; then :
         JSONC_LIBS=""
         JSONC_CFLAGS=""
     ])
-    AX_PTHREAD([],[
-        AC_MSG_WARN([POSIX Threads not available. Building library only.])
-        JSONC_LIBS=""
-        JSONC_CFLAGS=""
-        EXTRA_TARGETS=""
+    dnl> AX_PTHREAD test does not work along with MingW/MSYS anymore
+    AS_IF([test "x$BUILD_MINGW" != "x1"],[
```
Comment (IvanNardi) [inline configure.ac]: Not sure how this stuff works, but... `BUILD_MINGW` is set only **later** around line 285... My 2 cents

Comment (utoni) [reply configure.ac]: Thanks, I was kinda lost here. Didn't see the obvious..

**Pull request merged**